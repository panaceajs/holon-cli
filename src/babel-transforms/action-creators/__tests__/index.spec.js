const pluginTester = require('babel-plugin-tester');
const actionCreators = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: actionCreators,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title: 'it should add nothing when there is nothing to add.',
      fixture: '__fixtures__/without-exports.js',
      snapshot: false
    },
    {
      title: 'it should add action creators.',
      fixture: '__fixtures__/without-exports.js',
      pluginOptions: { actionTypes: { foo: 'foo' } }
    },
    {
      title: 'it should add action creator exports.',
      fixture: '__fixtures__/without-exports.js',
      pluginOptions: {
        targetPath: '../../actions',
        actionTypes: { foo: 'foo' }
      }
    },
    {
      title: 'it should update action creator exports.',
      fixture: '__fixtures__/with-exports.js',
      pluginOptions: {
        targetPath: '../actions',
        actionTypes: {
          existingAction: 'existingAction',
          nonExistingAction: 'nonExistingAction'
        }
      }
    }
  ]
});
