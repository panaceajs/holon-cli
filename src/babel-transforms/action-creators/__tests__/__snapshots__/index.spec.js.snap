// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-action-creators it should add action creator exports.: it should add action creator exports. 1`] = `
"
console.log('nothing going on here');

      ↓ ↓ ↓ ↓ ↓ ↓

console.log('nothing going on here');
export const foo = createAction(FOO);
"
`;

exports[`holon-action-creators it should add action creators.: it should add action creators. 1`] = `
"
console.log('nothing going on here');

      ↓ ↓ ↓ ↓ ↓ ↓

console.log('nothing going on here');
export const foo = createAction(FOO);
"
`;

exports[`holon-action-creators it should update action creator exports.: it should update action creator exports. 1`] = `
"
export const existingAction = createAction(EXISTING_ACTION);
console.log('nothing going on here');

      ↓ ↓ ↓ ↓ ↓ ↓

export const existingAction = createAction(EXISTING_ACTION);
console.log('nothing going on here');
export const nonExistingAction = createAction(NON_EXISTING_ACTION);
"
`;
